packages/shared/src/hooks/bookmark/useCreateBookmarkFolder.ts

```diff
@@ -0,0 +1,52 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import { BookmarkFolder, createBookmarkFolder } from '../../graphql/bookmarks';
+import { useLogContext } from '../../contexts/LogContext';
+import { LogEvent } from '../../lib/log';
+import { useToastNotification } from '../useToastNotification';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+
+type CreateBookmarkFolderProps = Pick<BookmarkFolder, 'name' | 'icon'>;
+
+interface UseCreateBookmarkFolder {
+  isPending: boolean;
+  createFolder: (folder: CreateBookmarkFolderProps) => void;
+}
+
+export const useCreateBookmarkFolder = (): UseCreateBookmarkFolder => {
+  const { logEvent } = useLogContext();
+  const { displayToast } = useToastNotification();
+  const queryClient = useQueryClient();
+
+  const { mutateAsync, isPending } = useMutation({
+    mutationFn: createBookmarkFolder,
+  });
+
+  const createFolder = useCallback(
+    async (folder: CreateBookmarkFolderProps) => {
+      await mutateAsync(folder)
+        .then(async ({ id }) => {
+          logEvent({
+            event_name: LogEvent.CreateBookmarkFolder,
+            target_id: id,
+          });
+
+          const listQueryKey = generateQueryKey(RequestKey.BookmarkFolders);
+          queryClient.setQueryData(listQueryKey, (data: BookmarkFolder[]) => {
+            return [...data, { id, ...folder }];
+          });
+
+          displayToast(`${folder.name} has been created`);
```

Comment:
This should be conditional. When creating a new folder through the "move folder" modal, we probably don't want to display this toast, because we will be displaying the "Moved to X" toast instead

Reply:
This hook only handle creation, update will be handled in folder page so we don't need to handle this case

Reply:
@ilasw There is a "create" button in the choose folder modal:
![Screenshot 2024-12-09 at 12 42 20](https://github.com/user-attachments/assets/d729622a-8010-4b39-a2ce-7fbeee96e975)

Reply:
[This](https://dailydotdev.slack.com/archives/C080TG5FHJB/p1733241538244629?thread_ts=1733150942.315109&cid=C080TG5FHJB) should be the flow, so we create the list first then we assign to the new folder. No need to change the message imo 👀